### PR TITLE
Issue #12171: Streaming Window FILTER

### PR DESCRIPTION
**Pull request by hawkfish**
* Add filtered aggregate support in PhysicalStreamingWindow.
* Refactor to share data structures between calls.
* Disable streaming when the optimiser is off so we can check results against the non-streaming operator.

### CODE REVIEW

File: src/execution/operator/aggregate/physical_streaming_window.cpp [hunk 1]
```diff
@@ -50,41 +50,60 @@ class StreamingWindowState : public OperatorState {
 public:
 	using StateBuffer = vector<data_t>;
 
-	StreamingWindowState()
-	    : initialized(false), allocator(Allocator::DefaultAllocator()),
-	      statev(LogicalType::POINTER, data_ptr_cast(&state_ptr)) {
-	}
+	struct AggregateState {
```
Comment (lnkuiper) [inline src/execution/operator/aggregate/physical_streaming_window.cpp:53]: Won't this name conflict with `struct AggregateState` in `physical_ungrouped_aggregate.cpp` in our amalgamation builds?

**Review by lnkuiper — APPROVED** [review]
Looks great, thanks! I just have one question:

File: src/execution/operator/aggregate/physical_streaming_window.cpp (hunk 1 above)
Comment (hawkfish) [reply src/execution/operator/aggregate/physical_streaming_window.cpp:53]: I don't think so - it is an inner class.

**Comment by hawkfish** [comment]
Sorry pushed change on the wrong branch.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**